### PR TITLE
Enable ES Tests on Master

### PR DESCRIPTION
**Pull request by noronhaa**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: charts/ccd-data-store-api/values.preview.template.yaml [hunk 1]
```diff
@@ -16,12 +16,12 @@ java:
...
     DATA_STORE_DB_OPTIONS: "?stringtype=unspecified"
     DATA_STORE_DB_MAX_POOL_SIZE: 10
-    DEFINITION_STORE_HOST: http://ccd-definition-store-api-pr-575.service.core-compute-preview.internal/
+    DEFINITION_STORE_HOST: http://ccd-definition-store-api-PR-891.service.core-compute-preview.internal/
     USER_PROFILE_HOST: http://ccd-user-profile-api-pr-335.service.core-compute-preview.internal/
     ENABLE_DB_MIGRATE: true
 
-    ELASTIC_SEARCH_ENABLED: false # enable whenever ES required on a particular PR
-    ELASTIC_SEARCH_NODES_DISCOVERY_ENABLED: false
+    ELASTIC_SEARCH_ENABLED: true # enable whenever ES required on a particular PR
```
Comment (mario-paniccia) [inline charts/ccd-data-store-api/values.preview.template.yaml:23]: this means we are enabling ES func tests on every PR now?

File: src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java [hunk 2]
```diff
@@ -25,13 +26,32 @@
...
+            .findFirst()
+            .map(t -> t.substring(1))
+            .map(Object::toString)
+            .orElse("error cant find tag");
+    }
+
     @Override
     public void doLoadTestData() {
+        if (elasticSearchEnabled()) {
+            new ElasticSearchUtils().deleteIndexesIfPresent();
```
Comment (mario-paniccia) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java:53]: let's have a chat about pros and cons of deleting indexes. My fear is that AAT should represent production, and in Production we don't always delete indexes when releasing changes. So if on AAT we always do, we risk of not catching potential issues on AAT but then seeing them on Prod...

File: src/aat/java/uk/gov/hmcts/ccd/datastore/befta/ElasticSearchUtils.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        asElasticsearchApiUser().when().delete(getCaseIndexAliasApi(AAT_PRIVATE_INDEX_NAME, AAT_PRIVATE_INDEX_ALIAS));
+        asElasticsearchApiUser().when().delete(AAT_PRIVATE_INDEX_NAME);
+        asElasticsearchApiUser().when().delete(getCaseIndexAliasApi(AAT_PRIVATE2_INDEX_NAME, AAT_PRIVATE2_INDEX_ALIAS));
+        asElasticsearchApiUser().when().delete(AAT_PRIVATE2_INDEX_NAME);
+        asElasticsearchApiUser().when().delete(getCaseIndexAliasApi(BEFTA_JURISDICTION_2_CASETYPE_1_NAME,
+            BEFTA_JURISDICTION_2_CASETYPE_1_ALIAS));
+        asElasticsearchApiUser().when().delete(BEFTA_JURISDICTION_2_CASETYPE_1_NAME);
+        asElasticsearchApiUser().when().delete(getCaseIndexAliasApi(FT_COMPLEXCOLLECTIONCOMPLEX_INDEX_NAME,
+            FT_COMPLEXCOLLECTIONCOMPLEX_INDEX_ALIAS));
+        asElasticsearchApiUser().when().delete(FT_COMPLEXCOLLECTIONCOMPLEX_INDEX_NAME);
```
Comment (mario-paniccia) [inline src/aat/java/uk/gov/hmcts/ccd/datastore/befta/ElasticSearchUtils.java:38]: what indexes are these? these are not all the BEFTA case types indexes right?

File: src/aat/resources/features/F-104/S-635.td.json [hunk 4]
```diff
@@ -142,7 +143,8 @@
             "[SECURITY_CLASSIFICATION]": "PRIVATE",
             "[CASE_TYPE]": "AAT_PRIVATE",
             "[LAST_MODIFIED_DATE]": "[[ANYTHING_PRESENT]]"
-          }
+          },
+          "supplementary_data" : null
```
Comment (mario-paniccia) [inline src/aat/resources/features/F-104/S-635.td.json:147]: `"supplementary_data" : null`  is repeated quite often in the various test files. Is it possible to move it up one level so it's inherited everywhere? I think @Abi-Laver worked on a similar enhancement she would know

File: src/aat/java/uk/gov/hmcts/ccd/datastore/befta/ElasticSearchUtils.java (hunk 3 above)
Comment (noronhaa) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/befta/ElasticSearchUtils.java:38]: No not all of them, these are a sub section of befta indexes that have tests that fail when the index isn't empty. ie it pulls back cases from previous test runs as well as the current test run when we are only expecting a result from the current test run. Based on your other comment may need to think about another solution to this

File: charts/ccd-data-store-api/values.preview.template.yaml (hunk 1 above)
Comment (noronhaa) [reply charts/ccd-data-store-api/values.preview.template.yaml:23]: This is mainly for testing this PR only, Tests on PR's wont work 'out of the box' so I would suggest either 
this gets set to false before merge or we follow up with another pr to set ELASTIC_SEARCH_ENABLED: false after master build is green

File: src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java (hunk 2 above)
Comment (noronhaa) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java:53]: ok

File: charts/ccd-data-store-api/values.preview.template.yaml (hunk 1 above)
Comment (mario-paniccia) [reply charts/ccd-data-store-api/values.preview.template.yaml:23]: ok got it cheers

File: src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java (hunk 2 above)
Comment (mario-paniccia) [reply src/aat/java/uk/gov/hmcts/ccd/datastore/befta/DataStoreTestAutomationAdapter.java:53]: spoke to Ashely, we'll keep this for now as a tactical solution, but we'll start working on the strategic one of using "contains" to verify results

**Review by danlysiak — APPROVED** [review]

**Review by RebeccaBaker — APPROVED** [review]

**Pull request merged**